inst/shiny/DiagnosticsExplorer/R/FacetGrid2.R

```diff
@@ -0,0 +1,443 @@
+#'     \item{`"all"` or `TRUE`}{Both x- and y-scales are allowed to vary within
+#'     a column or row respectively.}
+#'   }
+#' @param strip An object created by a call to a strip function, such as
+#'   [`strip_vanilla`][strip_vanilla()].
+#'
+#' @details Both the `independent` and `space` arguments only have an effect
+#'   when the `scales` argument in a dimension is free. However, the
+#'   `independent` and `space` arguments can *not* be used to simultaneously set
+#'   an independent scale and have the panel size be proportional to that scale.
+#'
+#' @family facetting functions
+#' @return A `Facet` ggproto object that can be added to a plot.
+#' @export
+#' @md
+#'
+#' @examples
+#' p <- ggplot(mpg, aes(displ, hwy)) + geom_point()
+#'
+#' # Repeat all axes for every facet
+#' p + facet_grid2(cyl ~ drv, axes = "all")
+#'
+#' # Repeat only y-axes
+#' p + facet_grid2(cyl ~  drv, axes = "y")
+#'
+#' # Repeat axes without x-labels
+#' p + facet_grid2(cyl ~ drv, axes = "all", remove_labels = "x")
+#'
+#' # Grid facets with independent axes for every panel
+#' p + facet_grid2(cyl ~ drv, scales = "free", independent = "all")
+facet_grid2 <- function(
+    rows = NULL,
+    cols = NULL,
+    scales = "fixed",
+    space  = "fixed",
+    axes   = "margins",
+    remove_labels = "none",
+    independent = "none",
+    shrink   = TRUE,
+    labeller = "label_value",
+    as.table = TRUE,
+    switch  = NULL,
+    drop    = TRUE,
+    margins = FALSE,
+    strip = strip_vanilla()
+) {
+  new_grid_facets(
+    rows, cols,
+    scales, space, axes, remove_labels, independent,
+    shrink, labeller, as.table, switch,
+    drop, margins, strip,
+    super = FacetGrid2
+  )
+}
+
+# Internal ----------------------------------------------------------------
+
+new_grid_facets <- function(
+    rows, cols,
+    scales, space, axes, rmlab, indy,
+    shrink, labeller, as.table, switch,
+    drop, margins, strip,
+    params = list(), super = FacetGrid2
+) {
+  # Check arguments
+  switch <- switch %||% "none"
+  switch <- arg_match0(switch, c("none", "both", "x", "y"))
+  labeller <- check_labeller(labeller)
+  axes  <- .match_facet_arg(axes,   c("margins", "x", "y", "all"))
+  free  <- .match_facet_arg(scales, c("fixed", "free_x", "free_y", "free"))
+  space <- .match_facet_arg(space,  c("fixed", "free_x", "free_y", "free"))
+  rmlab <- .match_facet_arg(rmlab,  c("none", "x", "y", "all"))
+  indy  <- .match_facet_arg(indy,   c("none", "x", "y", "all"))
+  strip <- assert_strip(strip)
+  
+  # Validate axes drawing parameters
+  axis_params <- .validate_independent(indy, free, space, rmlab)
+  
+  # Setup facet variables
+  facets <- .int$grid_as_facets_list(rows, cols)
+  
+  # Make list of parameters
+  params <- c(params, axis_params, list(
+    rows = facets$rows,
+    cols = facets$cols,
+    margins = margins,
+    labeller = labeller,
+    as.table = as.table,
+    switch = switch,
+    drop = drop,
+    axes = axes
+  ))
+  
+  ggproto(
```

Comment:
please use calls to external libraries using the `::` namespace (e.g. `ggplot2::ggproto`). 
This is because we don't want to create problems by calling `library` or `require` which will overwrite anything in the user's R session.